Add veracode workflow

anaarmas opened the pull request:
<!--
IMPORTANT:

This repository contains configuration for what users see when they click on the `Actions` tab and the setup page for Code Scanning.

It is not:
* A playground to try out scripts
* A place for you to create a workflow for your repository
-->

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

anaarmas commented on code-scanning/veracode.yml:
```diff
@@ -0,0 +1,60 @@
+
+    - run: zip -r veracode-pipeline-scan-results-to-sarif.zip ./
+
+    - uses: actions/upload-artifact@v1
```
I'm not sure what the purpose of this action is within the workflow and I'm concerned that it will results in unnecessarily consuming users' artifact storage.

cc @josepalafox so we can get Veracode's input here.

Lerer replied:
You are correct - this step is not required for the Veracode Pipeline-Scan workflow.

anaarmas commented on code-scanning/veracode.yml:
```diff
@@ -0,0 +1,60 @@
+        java-version: 1.8
+    - run: java -jar pipeline-scan.jar --veracode_api_id "${{secrets.VERACODE_API_ID}}" --veracode_api_key "${{secrets.VERACODE_API_KEY}}" --fail_on_severity="Very High, High" --file veracode-pipeline-scan-results-to-sarif.zip
+      continue-on-error: true
+    - uses: actions/upload-artifact@v1
```
same as above

Lerer replied:
The `results.json` file is the raw output (in JSON format) of the Veracode Pipeline-Scan.

The step was originally added to provide users with a reference to the pre-transformed copy of the scan results.
It does not effect or contribute to the functionality of the workflow

mehran-sayyah commented:
ok

Lerer commented on code-scanning/veracode.yml:
```diff
@@ -0,0 +1,60 @@
+      with:
+        repository: ''
+
+    - run: zip -r veracode-pipeline-scan-results-to-sarif.zip ./
```
a better archive (zip file) name would be: `veracode-scan-target.zip`

Also need to follow with the change in the line of the scan command: 46

mehran-sayyah replied:
46

Lerer replied:
I meant, if we can change the zip file name here, then we should also change the name in line 46

fUzje7-butgyc-kejzyx commented:
Le ven. 3 déc. 2021 à 00:46, مهران سیاح ***@***.***> a écrit :

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In code-scanning/veracode.yml
> <https://github.com/actions/starter-workflows/pull/1278#discussion_r761661775>
> :
>
> > +    - cron: $cron-weekly
> +
> +# A workflow run is made up of one or more jobs that can run sequentially or in parallel
> +jobs:
> +  # This workflow contains a job to build and submit pipeline scan, you will need to customize the build process accordingly and make sure the artifact you build is used as the file input to the pipeline scan file parameter
> +  build-and-pipeline-scan:
> +    # The type of runner that the job will run on
> +    runs-on: ubuntu-latest
> +    steps:
> +
> +    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it and copies all sources into ZIP file for submitting for analysis. Replace this section with your applications build steps
> +    - uses: ***@***.***
> +      with:
> +        repository: ''
> +
> +    - run: zip -r veracode-pipeline-scan-results-to-sarif.zip ./
>
> 46
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/actions/starter-workflows/pull/1278#discussion_r761661775>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUEIUPFL2QK3AHM3DX6QKYDUPBKSZANCNFSM5JBHCMYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

mehran-sayyah commented:
#1278

دریافت Outlook برای Android<https://aka.ms/AAb9ysg>
________________________________
From: Coby Lerer ***@***.***>
Sent: Saturday, December 4, 2021 6:29:06 AM
To: actions/starter-workflows ***@***.***>
Cc: مهران سیاح ***@***.***>; Comment ***@***.***>
Subject: Re: [actions/starter-workflows] Add veracode workflow (PR #1278)


@Lerer commented on this pull request.

________________________________

In code-scanning/veracode.yml<https://github.com/actions/starter-workflows/pull/1278#discussion_r762376629>:

> +    - cron: $cron-weekly
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a job to build and submit pipeline scan, you will need to customize the build process accordingly and make sure the artifact you build is used as the file input to the pipeline scan file parameter
+  build-and-pipeline-scan:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    steps:
+
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it and copies all sources into ZIP file for submitting for analysis. Replace this section with your applications build steps
+    - uses: ***@***.***
+      with:
+        repository: ''
+
+    - run: zip -r veracode-pipeline-scan-results-to-sarif.zip ./


I meant, if we can change the zip file name here, then we should also change the name in line 46

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/actions/starter-workflows/pull/1278#discussion_r762376629>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATTU54CCKSNSE4BGSKDV4ZTUPF7XVANCNFSM5JBHCMYA>.

anaarmas commented:
@Lerer thanks for your quick reply! I've removed the two uses of the upload-artifact action and I've also renamed the input file as you suggested. @nickfyson I think this is ready to merge!

nickfyson reviewed (approved):
🚀
